silkrpc/ethbackend/backend.hpp

```diff
@@ -189,6 +220,192 @@ class BackEnd final {
+        silkworm::Bloom bloom;
+        std::memcpy(&bloom[0], bytes_from_H2048(logs_bloom_h2048).data(), 256);
+        // Convert transactions in std::string to silkworm::Bytes
+        std::vector<silkworm::Bytes> transactions;
+        for (const auto& transaction_string : execution_payload_grpc.transactions()) {
+            transactions.push_back(silkworm::bytes_of_string(transaction_string));
+        }
+
+        // Assembling the execution_payload data structure
+        return ExecutionPayload{
+            .number = execution_payload_grpc.blocknumber(),
+            .timestamp = execution_payload_grpc.timestamp(),
+            .gas_limit = execution_payload_grpc.gaslimit(),
+            .gas_used = execution_payload_grpc.gasused(),
+            .suggested_fee_recipient = address_from_H160(execution_payload_grpc.coinbase()),
+            .state_root = bytes32_from_H256(state_root_h256),
+            .receipts_root = bytes32_from_H256(receipts_root_h256),
+            .parent_hash = bytes32_from_H256(parent_hash_h256),
+            .block_hash = bytes32_from_H256(block_hash_h256),
+            .random = bytes32_from_H256(random_h256),
+            .base_fee = uint256_from_H256(base_fee_h256),
+            .logs_bloom = bloom,
+            .extra_data = silkworm::bytes_of_string(extra_data_string),
+            .transactions = transactions
+        };
+    }
+
+    types::ExecutionPayload encode_execution_payload_to_grpc_format(const ExecutionPayload& execution_payload) {
+        types::ExecutionPayload execution_payload_grpc;
+        // Numerical parameters
+        execution_payload_grpc.set_blocknumber(execution_payload.number);
+        execution_payload_grpc.set_timestamp(execution_payload.timestamp);
+        execution_payload_grpc.set_gaslimit(execution_payload.gas_limit);
+        execution_payload_grpc.set_gasused(execution_payload.gas_used);
+        // coinbase
+        execution_payload_grpc.set_allocated_coinbase(H160_from_address(execution_payload.suggested_fee_recipient));
+        // 32-bytes parameters
+        execution_payload_grpc.set_allocated_receiptroot(H256_from_bytes(execution_payload.receipts_root.bytes));
+        execution_payload_grpc.set_allocated_stateroot(H256_from_bytes(execution_payload.state_root.bytes));
+        execution_payload_grpc.set_allocated_parenthash(H256_from_bytes(execution_payload.parent_hash.bytes));
+        execution_payload_grpc.set_allocated_random(H256_from_bytes(execution_payload.random.bytes));
+        execution_payload_grpc.set_allocated_basefeepergas(H256_from_uint256(execution_payload.base_fee));
+        // Logs Bloom
+        execution_payload_grpc.set_allocated_logsbloom(H2048_from_bytes(&execution_payload.logs_bloom[0]));
+        // String-like parameters
+        for (auto transaction_bytes : execution_payload.transactions) {
+            execution_payload_grpc.add_transactions(std::string(reinterpret_cast<char*>(&transaction_bytes[0]), transaction_bytes.size()));
```

Comment:
You should consider `std::string{transaction_bytes.begin(), transaction_bytes.end()}` instead of `std::string(reinterpret_cast<char*>(&transaction_bytes[0]), transaction_bytes.size())` as it is shorter, avoids a cast and is IMHO more readable.